docs/using/libraries/node/index.rst

```diff
@@ -0,0 +1,196 @@
+=============================
+Node.js Authorization Library
+=============================
+
+oso is packaged as a :doc:`npm module </download>` for use in Node.js
+applications.
+
+API documentation lives :doc:`here</js/node/index>`.
+
+.. toctree::
+    :hidden:
+
+    /js/node/index
+
+To install, see :doc:`installation instructions </download>`.
+
+.. todo:: typescript
+
+Working with JavaScript Types
+=============================
+
+oso's Node.js authorization library allows you to write policy rules over
+JavaScript types directly. This document explains how different types of
+JavaScript values can be used in oso policies.
+
+.. note::
+  More detailed examples of working with application objects can be found in :doc:`/using/examples/index`.
+
+Objects
+^^^^^^^
+
+You can pass any JavaScript object into oso and access its properties from
+your policy (see :ref:`application-types`).
+
+Class Instances
+^^^^^^^^^^^^^^^
+
+Any ``new``-able JavaScript object (including ES6-style classes) can be
+constructed from inside an oso policy using the :ref:`operator-new` operator if
+the constructor (a ``class`` or ``function`` that responds to `the new operator
+<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new>`_)
+has been **registered** using the ``#registerClass`` method. An example of
```

Comment:
Is the `#` a JS convention also, or is this left over from ruby?

Reply:
it's not a JS convention; we could drop it

Reply:
It's not used in other places to refer to JS methods (that I saw), so maybe best to drop it.

Reply:
agree; follow-up